src/types/routeProps.ts

```diff
@@ -30,10 +30,24 @@ type WithHooks = {
   onAfterRouteLeave?: MaybeArray<AfterRouteHook>,
 }
 
+type WithComponent = {
+  /**
+   * A Vue component, which can be either synchronous or asynchronous components.
+   */
+  component: Component,
+}
+
+type WithComponents = {
+  /**
+   * Multiple components for named views, which can be either synchronous or asynchronous components.
+   */
+  components: Record<string, Component>,
+}
+
 /**
  * Represents properties common to parent routes in a route configuration, including hooks, path, and optional query parameters.
  */
-export type ParentRouteProps = WithHooks & {
+export type ParentRouteProps = Partial<WithComponent | WithComponents> & WithHooks & {
```

Comment:
Wrote this in a way where you can define a `component` property or a `components` property but not both. Which means if you want to do anything with either of those properties you have to type guard it first to determine which one is available. 

I think this is a good devx when defining routes using `createRoutes`. But its a little annoying if you need to access the component from `route.match.component` or `route.match.components` because you have to type guard it. But I don't expect users to be accessing those properties. So I think its safe to take on that slight annoyance in our internal codebase to give users a good devx when defining routes.